### PR TITLE
Feature: Form Fields onChange handling

### PR DESCRIPTION
**Pull request by Drish-xD**
Changes Done 

- Handle Preset Appling efficiently
- Handle onChange group, setting parentBatch and subBatch options
- Handle Popup and SignUp fields conditionally show 
- Minor package upgrades
- Logo updated

### CODE REVIEW

File: src/app/session/[type]/Steps/helper.ts [hunk 1]
```diff
@@ -9,130 +9,142 @@ import {
...
+  if (!value) return;
 
+  const authGroupId = apiOptions.group?.find((item) => item.value === value)?.id;
   const filteredQuizBatchOptions = apiOptions?.batch?.filter(
     (item) => item.groupId === authGroupId && !item.parentId
   );
 
   (fieldsSchema.parentBatch as MySelectProps).options = filteredQuizBatchOptions ?? [];
   (fieldsSchema.subBatch as MySelectProps).options = [];
+  console.log('[OPTIONS] PARENT BATCH APPLIED : ', value);
```
Comment (suryabulusu) [inline src/app/session/[type]/Steps/helper.ts:147]: remove

File: src/app/session/[type]/Steps/helper.ts [hunk 2]
```diff
@@ -141,16 +153,20 @@ export const setBatchOptions = (
...
-  const quizBatchId = apiOptions.batch?.find((item) => item.value === value)?.id;
+  if (!value) return;
 
+  const quizBatchId = apiOptions.batch?.find((item) => item.value === value)?.id;
   const filteredClassBatchOptions = apiOptions?.batch?.filter(
     (item) => item.parentId === quizBatchId
   );
   form.setValue('subBatch', '');
   (fieldsSchema.subBatch as MySelectProps).options = filteredClassBatchOptions ?? [];
+  console.log('[OPTIONS] SUB BATCH APPLIED : ', value);
```
Comment (suryabulusu) [inline src/app/session/[type]/Steps/helper.ts:164]: remove

**Review by suryabulusu — COMMENTED** [review]
looks good.. remove console log

**Review by suryabulusu — APPROVED** [review]

File: src/app/session/[type]/Steps/helper.ts (hunk 1 above)
Comment (Drish-xD) [reply src/app/session/[type]/Steps/helper.ts]: removed

**Pull request merged**